### PR TITLE
Add check input focus style

### LINKED ISSUE #28957: No focus style on checkbox and radios
### Description

There's no visual feedback when focusing a checkbox or radio.

### Ask

- Is this intended ?
- If not, will it be fixed ?

### Test case

Can be tested on [the current doc](https://getbootstrap.com/docs/4.3/components/forms/#checkboxes-and-radios)

### Additional Info

- A similar bug was opened for v.3 and has been fixed #14923
- [Custom form controls](https://getbootstrap.com/docs/4.3/components/forms/#custom-forms) checkboxes and radios have a focus style.

### PR DESCRIPTION
**Pull request by alienlebarge**
### Description

Add focus style for checkboxes and radios

### Additional Info

1. Initial Issue: close #28957 
1. It was first propose tu make the change for current and next version of bootstrap. But it was decided to only target v4. (See #28959 for more info)

Demo: https://deploy-preview-28975--twbs-bootstrap.netlify.com/docs/4.3/components/forms/#without-labels

### CODE REVIEW

**Comment by patrickhlauke** [comment]
@XhmikosR this probably needs some kind of squashing/rebasing/cherry-picking i think ... the change itself is just to that one scss file but because it started out against master it carries a lot of baggage...my git-fu is not quite there, so you ok tackling that?

**Comment by alienlebarge** [comment]
If you prefer I can start from `v4-dev`

**Comment by MartijnCuppens** [comment]
We need to fix the border radius of the checkbox and radio somehow. The problem is the radius of the checkbox cannot be detected since not all browsers/OS'es have the same radius.

**Comment by patrickhlauke** [comment]
> We need to fix the border radius of the checkbox and radio somehow. The problem is the radius of the checkbox cannot be detected since not all browsers/OS'es have the same radius.

checkboxes and radio buttons themselves are stubborn (in Chrome in particular) against styling their border or anything, and the box shadow can't be influenced either it seems. the only way to get the rounded indication here would be to essentially redo the whole checkbox/radio input with faked elements, which is outside of the scope of this tiny change. adding a `border-radius` for radio type inputs does seem to have an effect in IE/Edge/Firefox though

i'd say we can live with the non-rounded focus indication in some browsers (Chrome for instance ... in Firefox, it's actually reasonably pleasing, particularly for the radio button), as this is a major step forward compared to some browsers (Firefox, IE, ...) not having a clear focus indication at all

File: scss/_forms.scss
Comment (patrickhlauke) [inline scss/_forms.scss]: tried adding a commit to this directly, but seems it may not be accepting commits from maintainers...

in any case, I'd suggest adding the following here to mitigate the slightly unsightly look of round radio buttons with a completely square box shadow in IE/Edge (Firefox already rounds them, and Chrome is just stubborn about any styling of checkboxes/radio buttons in general)

```
  &[type="radio"] {
    // Note: This has no effect on the appearance of the radio button, but does smooth out the box shadow on focus in most browsers.
    @include border-radius($input-border-radius, 0);
  }
```

**Review by patrickhlauke — CHANGES_REQUESTED** [review]

File: scss/_forms.scss
Comment (alienlebarge) [reply scss/_forms.scss]: I made [the change](https://github.com/alienlebarge/bootstrap/commit/565177310f939c6d452971649c115d18edf07f18) on the [add-check-input-focus-style-v4](https://github.com/alienlebarge/bootstrap/tree/add-check-input-focus-style-v4) branch on [alienlebarge/bootstrap](https://github.com/alienlebarge/bootstrap) but it doesn't appear in the PR...

Do you have any idea why ?

Comment (alienlebarge) [reply scss/_forms.scss]: @patrickhlauke done

**Comment by patrickhlauke** [comment]
for info @twbs/css-review not sure what that travis problem is, but seems unrelated to the CSS change here

**Review by patrickhlauke — APPROVED** [review]

**Comment by patrickhlauke** [comment]
noting that if the change is still felt too disruptive (because yes, in Chrome it currently doesn't look super fantastic), this could be flipped to use `:focus-visible` instead (which works in Firefox right now, and will be coming to Chrome and Edge ... so a future-facing change for when those browsers support it, and in the meantime it fixes the issue in Firefox and maintains the status quo in other browsers)

**Comment by XhmikosR** [comment]
@twbs/css-review: what should we do with this? It targets v4-dev BTW.

**Comment by MartijnCuppens** [comment]
I personally don't really like it. It just doesn't look that good:
image & image

**Comment by XhmikosR** [comment]
Your guys' call, feel free to close it. Looking at the remaining v4-dev stuff.

**Comment by XhmikosR** [comment]
@twbs/css-review final call for action about this PR?

**Comment by MartijnCuppens** [comment]
Not a fan of the current theming (see https://github.com/twbs/bootstrap/pull/28975#issuecomment-512275014), however I have no clue how I could improve it, so it's fine by me if it gets merged.

File: scss/_forms.scss
Comment (XhmikosR) [inline scss/_forms.scss]: @MartijnCuppens should this be here?

**Comment by ysds** [comment]
I have the same opinion as @MartijnCuppens

**Pull request closed without merging**
